src/docfx/publish/PublishUrlMap.cs

```diff
@@ -80,34 +80,41 @@ public IEnumerable<FilePath> GetFilesByUrl(string url)
             return default;
         }
 
-        private Dictionary<string, List<PublishUrlMapItem>> Initialize()
+        private (FilePath[] files, Dictionary<string, List<PublishUrlMapItem>> urlMap) Initialize()
         {
-            var builder = new ListBuilder<PublishUrlMapItem>();
-
             using (Progress.Start("Building publish url map"))
             {
-                Parallel.Invoke(
-                    () => ParallelUtility.ForEach(
-                        _errors, _redirectionProvider.Files.Where(x => x.Origin != FileOrigin.Fallback), file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(
-                        _errors,
-                        _buildScope.GetFiles(ContentType.Resource).Where(x => x.Origin != FileOrigin.Fallback || _config.OutputType == OutputType.Html),
-                        file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(
-                        _errors, _buildScope.GetFiles(ContentType.Page).Where(x => x.Origin != FileOrigin.Fallback), file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(_errors, _tocMap.GetFiles(), file => AddItem(builder, file)));
+                return CreateUrlMap(
```

Comment:
After this change, `toc` is not part of `urlMap` anymore?

Reply:
Right now, toc is not using `urlMap`. It might use in the future.

Reply:
yep.